Formula/sonixflasher.rb

```diff
@@ -0,0 +1,18 @@
+class Sonixflasher < Formula
+  desc "CLI-based Flasher for Sonix 24x/26x MCUs"
+  homepage "https://github.com/SonixQMK/SonixFlasherC"
+  # Use the tag instead of the tarball to get submodules
```

Comment:
The submodule is for the source of `hidapi`, we should use Homebrew `hidapi` instead: https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae

Reply:
> 

Can I override this in the formula without overhauling the makefile? I'm kinda new to the homebrew ecosystem. Thanks!

Reply:
It is possible, but probably not something we'll accept for a new formula since doing this is rather fragile.

It would be better to send a pull request to the upstream repository that allows use of a system-provided `hidapi` (e.g. found via `pkg-config` much like what is done with `libudev` on Linux) if available.

Reply:
requested changes submitted